### ISSUE TITLE
Cannot install logstash-input-websocket with 7.0.0-beta1

### ISSUE THREAD

**Issue by spinscale** [logstash-plugins/logstash-input-websocket#19]
For all general issues, please provide the following details for fast resolution:

- Version: 7.0.0-beta1
- Operating System: osx
- Steps to Reproduce:

```
#  ./bin/logstash-plugin install logstash-input-websocket
OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.jruby.ext.openssl.SecurityHelper (file:/Users/alr/Downloads/stacks/7.0.0-beta1/logstash-7.0.0-beta1/vendor/jruby/lib/ruby/stdlib/jopenssl.jar) to constructor java.security.cert.CertificateFactory(java.security.cert.CertificateFactorySpi,java.security.Provider,java.lang.String)
WARNING: Please consider reporting this to the maintainers of org.jruby.ext.openssl.SecurityHelper
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Validating logstash-input-websocket
Installing logstash-input-websocket
Plugin version conflict, aborting
ERROR: Installation Aborted, message: Bundler could not find compatible versions for gem "addressable":
  In snapshot (Gemfile.lock):
    addressable (= 2.4.0)

  In Gemfile:
    logstash-input-websocket java was resolved to 4.0.4, which depends on
      ftw (~> 0.0.46) java was resolved to 0.0.46, which depends on
        addressable (~> 2.2.0) java

    json-schema (~> 2) java was resolved to 2.8.1, which depends on
      addressable (>= 2.4) java

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.

Bundler could not find compatible versions for gem "logstash":
  In Gemfile:
    logstash-input-websocket java was resolved to 0.1.2, which depends on
      logstash (< 2.0.0, >= 1.4.0) java

Could not find gem 'logstash (< 2.0.0, >= 1.4.0)', which is required by gem 'logstash-input-websocket', in any of the sources.

Bundler could not find compatible versions for gem "logstash-core":
  In snapshot (Gemfile.lock):
    logstash-core (= 7.0.0)

  In Gemfile:
    logstash-core java

    logstash-core-plugin-api java was resolved to 2.1.16, which depends on
      logstash-core (= 7.0.0) java

    logstash-input-syslog java was resolved to 3.4.1, which depends on
      logstash-filter-grok java was resolved to 4.0.4, which depends on
        logstash-core (>= 5.6.0) java

    logstash-input-websocket java was resolved to 0.1.4, which depends on
      logstash-core (< 2.0.0, >= 1.4.0) java

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.

Bundler could not find compatible versions for gem "logstash-core-plugin-api":
  In snapshot (Gemfile.lock):
    logstash-core-plugin-api (= 2.1.16)

  In Gemfile:
    logstash-devutils (~> 1) java was resolved to 1.3.6, which depends on
      logstash-core-plugin-api (<= 2.99, >= 2.0) java

    logstash-input-websocket java was resolved to 3.0.2, which depends on
      logstash-core-plugin-api (~> 1.0) java

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
```

**Comment by magnusbaeck** [comment]
Like with #15, the problem is the ftw gem which only declares compatibility with addressable 2.3.x but other parts of Logstash require a more recent version of addressable. The ftw gem needs to be updated and I've asked about this in https://github.com/jordansissel/ruby-ftw/pull/38.

**Comment by conteit** [comment]
Any update on this issue? I have the same situation with logstash v7.0.1 and logstash-input-websocket v4.0.4.

**Comment by magnusbaeck** [comment]
The bad dependency in FTW was fixed in https://github.com/jordansissel/ruby-ftw/pull/43 and released in 0.0.49 of that gem so we're okay now. I verified that the plugin installs cleanly on Logstash 7.1.1 and 7.4.2.
